test/AutoDiff/downstream/activity_analysis.swift

```diff
@@ -568,3 +568,100 @@ func testBeginApplyActiveButInitiallyNonactiveInoutArgument(x: Float) -> Float {
 // CHECK: [NONE]   // function_ref Array.subscript.getter
 // CHECK: [NONE]   %34 = apply %33<Float>(%32, %29, %31) : $@convention(method) <τ_0_0> (Int, @guaranteed Array<τ_0_0>) -> @out τ_0_0
 // CHECK: [ACTIVE]   %35 = load [trivial] %32 : $*Float
+
+//===----------------------------------------------------------------------===//
+// Enum differentiation
+//===----------------------------------------------------------------------===//
+
+// expected-error @+1 {{function is not differentiable}}
+@differentiable
+// expected-note @+1 {{when differentiating this function definition}}
+func testActiveOptional(_ x: Float) -> Float {
+  // expected-note @+1 {{differentiating enum values is not yet supported}}
+  var maybe: Float? = 10
+  maybe = x
+  return maybe!
+}
+
+// CHECK-LABEL: [AD] Activity info for $s17activity_analysis12testOptionalyS2fF at (source=0 parameters=(0))
+// CHECK: bb0:
+// CHECK: [ACTIVE] %0 = argument of bb0 : $Float
+// CHECK: [ACTIVE]   %2 = alloc_stack $Optional<Float>, var, name "maybe"
+// CHECK: [USEFUL]   %3 = integer_literal $Builtin.IntLiteral, 10
+// CHECK: [USEFUL]   %4 = metatype $@thin Float.Type
+// CHECK: [NONE]   // function_ref Float.init(_builtinIntegerLiteral:)
+// CHECK: [USEFUL]   %6 = apply %5(%3, %4) : $@convention(method) (Builtin.IntLiteral, @thin Float.Type) -> Float
+// CHECK: [USEFUL]   %7 = enum $Optional<Float>, #Optional.some!enumelt.1, %6 : $Float
+// CHECK: [ACTIVE]   %9 = enum $Optional<Float>, #Optional.some!enumelt.1, %0 : $Float
+// CHECK: [ACTIVE]   %10 = begin_access [modify] [static] %2 : $*Optional<Float>
+// CHECK: [ACTIVE]   %13 = begin_access [read] [static] %2 : $*Optional<Float>
+// CHECK: [ACTIVE]   %14 = load [trivial] %13 : $*Optional<Float>
+// CHECK: bb1:
+// CHECK: [NONE]   %18 = integer_literal $Builtin.Word, 85
+// CHECK: [NONE]   %19 = integer_literal $Builtin.Int1, -1
+// CHECK: [NONE]   %20 = integer_literal $Builtin.Word, 537
+// CHECK: [NONE]   %21 = integer_literal $Builtin.Word, 15
+// CHECK: [NONE]   %22 = integer_literal $Builtin.Int1, 0
+// CHECK: [NONE]   // function_ref _diagnoseUnexpectedNilOptional(_filenameStart:_filenameLength:_filenameIsASCII:_line:_isImplicitUnwrap:)
+// CHECK: [NONE]   %24 = apply %23(%17, %18, %19, %20, %22) : $@convention(thin) (Builtin.RawPointer, Builtin.Word, Builtin.Int1, Builtin.Word, Builtin.Int1) -> ()
+// CHECK: bb2:
+// CHECK: [ACTIVE] %26 = argument of bb2 : $Float
+
+enum DirectEnum: Differentiable & AdditiveArithmetic {
+  case leaf(Float)
```

Comment:
Could you try one with multiple cases?

Reply:
I added more cases to `enum DirectEnum` and `enum IndirectEnum` in https://github.com/apple/swift/pull/29689/commits/02ed1383cb4e603f20aee71ea5ae83211d741d23.